discoveryApiResearchNowApiMapping.md

```diff
@@ -0,0 +1,436 @@
+# Discovery API - ResearchNow API mapping
+## Discovery API
+[Github](https://github.com/NYPL-discovery/discovery-api/blob/master/README.md#searching)
+### Parameters
+[GET `/v0.1/discovery/resources`](https://platformdocs.nypl.org/#/discovery/get_v0_1_discovery_resources)
+* `q` string
+* `page` string
+* `per_page` integer
+* `sort` integer
+  * "relevance", "title", "creator", "date"
+* `sort_direction` string
+  * "asc", "desc"
+  * title defaults to asc, date defaults to desc, creator defaults to asc, relevance is fixed desc
+* `search_scope` string
+  * "all", "title", "contributor", "subject", "series", "callnumber", "standard_number"
+* `filters` string
+  * "owner", "subjectLiteral", "holdingLocation", "deliveryLocation", "language", "materialType", "mediaType", "carrierType", "publisher", "contributor", "creator", "issuance", "createdYear", "dateAfte"', or "dateBefore"
+  * Specify a hash of filters to apply, where keys are from terms above
+
+## ResearchNow API
+[Github](https://github.com/NYPL/sfr-ingest-pipeline/tree/development/app/sfr-search-api#searching)
+### Parameters
+[GET `/v0.1/research-now/v3/search-api`](https://dev-platformdocs.nypl.org/#/research-now/get_v0_1_research_now_v3_search_api)
+* **`field`** (required) string
+  * "keyword", "title", "author", "standard_number" (ISBN, ISSN, LCCN and OCLC) and "subject"
+  * Defaults to "keyword"
+* **`query`** (required) string
+* `recordType`
+  * Internal record type to return with the work. Either instances or editions.
+* `page` integer
+* `per_page` integer
+* `sort`
+* `language`
+* `years`
+  * This should be formatted as {"start": year, "end": year}.
+
+<!-- <style>
+  table ul {
+    list-style: none;
+  }
+</style> -->
+
+## API Parameters Comparison
+This table lists, in the first column, the frontend features related to searching and search filters available from the frontend. The 2nd and 3rd column list out the Discovery API and ResearchNow API parameters that correlate to that frontend functionality.
+
+Single quotes (') are used for frontend terminology. Italics are used to describe frontend features. Code styling (e.g. `filters`) is used for the parameters the respective APIs except.
+
+|Discovery front end|Discovery API |ResearchNow API|
+|-------------------|--------------|---------------|
+|_Search field_     |`q`           |`query`        |
+|_Search field dropdown_|`search_scope`|`field`|
+|<ul>'All Fields'   |<ul>"all"     |<ul>"keyword"|
+|<ul>'Title'            |<ul>"title"        |<ul>"title"       |
+|<ul>'Author/Contributor'|<ul>"contributor" |<ul>"author"      |
+|<ul>'Standard Number'|<ul>"standard_number"|<ul> "standard_number"|
+|_Search page filters_|`filters`     |      |
+|<ul>'Format'       |<ul>`filters[materialType]`|<ul>N/A|
+|<ul>'Date':<ul><li>'Start Year'<li>'End Year'| <ul><br>`filters[dateAfter]`<br>`filters[dateBefore]` |<br><ul>`years[start]`<br>`years[end]`|
+|<ul>'Language'|<ul>`filters[language]`|<ul>`language`|
```

Comment:
The SFR filters also work in the same way as the SCC filters, as objects within a `filters` array, so a year filter would be `{'filters': [{'field': 'years', 'value': {'start': XXXX, 'end': XXXX}]}` and similar for language. Sorry if this was unclear in our documentation.

Reply:
There was a really helpful logging in the console yesterday that showed me that! Docs are a little unclear though.